idx/memory/memory.go

```diff
@@ -103,6 +103,29 @@ func (t *TagIndex) delTagId(name, value string, id idx.MetricID) {
 	}
 }
 
+// <name>;<key>=<name> -> Set of references to schema.MetricDefinition
+type defByTagSet map[string]map[*schema.MetricDefinition]struct{}
```

Comment:
why `*schema.MetricDefinition` and not `schema.MetricDefinition` ? would save a lot of pointers.

Reply:
If we copy the MDs then we'd need to make sure that all updates to MetricDefinitions are made to all instances of each. F.e. if we update `LastUpdate` we'd need to make sure that the update is done to all instances of that MetricDefinition, which seems to be complicated and prone to errors.